src/db.rs

```diff
@@ -41,41 +74,129 @@ impl AgateOptions {
         self
     }
 
-    pub fn wal_path<P: Into<PathBuf>>(&mut self, p: P) -> &mut AgateOptions {
-        self.wal_path = Some(p.into());
+    pub fn path<P: Into<PathBuf>>(&mut self, p: P) -> &mut AgateOptions {
+        self.path = p.into();
+        self
+    }
+
+    pub fn num_memtables(&mut self, num_memtables: usize) -> &mut AgateOptions {
+        self.num_memtables = num_memtables;
+        self
+    }
+
+    pub fn in_memory(&mut self, in_memory: bool) -> &mut AgateOptions {
+        self.in_memory = in_memory;
         self
     }
 
-    pub fn table_size(&mut self, size: u32) -> &mut AgateOptions {
-        self.table_size = size;
+    pub fn sync_writes(&mut self, sync_writes: bool) -> &mut AgateOptions {
+        self.sync_writes = sync_writes;
         self
     }
 
-    pub fn max_table_count(&mut self, count: usize) -> &mut AgateOptions {
-        self.max_table_count = count;
+    pub fn value_log_file_size(&mut self, value_log_file_size: u64) -> &mut AgateOptions {
+        self.value_log_file_size = value_log_file_size;
         self
     }
 
+    pub fn value_log_max_entries(&mut self, value_log_max_entries: u32) -> &mut AgateOptions {
+        self.value_log_max_entries = value_log_max_entries;
+        self
+    }
+
+    fn fix_options(&mut self) -> Result<()> {
+        if self.in_memory {
+            // TODO: find a way to check if path is set, if set, then panic with ConfigError
+            self.sync_writes = false;
+        }
+
+        Ok(())
+    }
+
     pub fn open<P: AsRef<Path>>(&mut self, path: P) -> Result<Agate> {
```

Comment:
can we move it to `struct Agate`?

Reply:
Implementing `open` in `XXXOptions` seems to be a convention. e.g. `OpenOptions` of `std::File`.

Reply:
Both badger and rocksdb follow the way of `DB::open(opt)`. If implementing in this way, it may cost more effect to integrate with engine trait with tikv.

Reply:
PTAL again

Reply:
Yes, I've already moved `open` from `AgateOptions` into `Agate`